src/plugins/visualizations/public/plugin.ts

```diff
@@ -400,8 +401,30 @@ export class VisualizationsPlugin
     uiActions.registerTrigger(dashboardVisualizationPanelTrigger);
     const editInLensAction = new EditInLensAction(data.query.timefilter.timefilter);
     uiActions.addTriggerAction(CONTEXT_MENU_TRIGGER, editInLensAction);
-    const embeddableFactory = new VisualizeEmbeddableFactory({ start });
-    embeddable.registerEmbeddableFactory(VISUALIZE_EMBEDDABLE_TYPE, embeddableFactory);
+    embeddable.registerReactEmbeddableFactory(VISUALIZE_EMBEDDABLE_TYPE, async () => {
+      const {
+        plugins: { embeddable: embeddableStart, embeddableEnhanced: embeddableEnhancedStart },
+      } = start();
+
+      const { getVisualizeEmbeddableFactory } = await import('./react_embeddable');
+      return getVisualizeEmbeddableFactory({ embeddableStart, embeddableEnhancedStart });
+    });
+    embeddable.registerReactEmbeddableSavedObject<VisualizationSavedObjectAttributes>({
+      onAdd: (container, savedObject) => {
+        container.addNewPanel<VisualizeSerializedState>({
```

Comment:
you still need to provided `initialState`. Add `initialState: { savedObjectId: savedObject.id },`

Reply:
It works without doing this for some reason

I'll add it but I'm confused